docs/buildrun.md

```diff
@@ -57,55 +57,55 @@ When the controller reconciles it:
 The `BuildRun` definition supports the following fields:
 
 - Required:
-  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `shipwright.io/v1alpha1`.
+  - [`apiVersion`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the API version, for example `shipwright.io/v1beta1`.
   - [`kind`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Specifies the Kind type, for example `BuildRun`.
   - [`metadata`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields) - Metadata that identify the CRD instance, for example the name of the `BuildRun`.
 
 - Optional:
-  - `spec.buildRef` - Specifies an existing `Build` resource instance to use. It cannot be used together with `buildSpec`.
-  - `spec.buildSpec` - Specifies an embedded (transient) Build resource to use. It cannot be used together with `buildRef`.
+  - `spec.build.name` - Specifies an existing `Build` resource instance to use.
+  - `spec.build.spec` - Specifies an embedded (transient) Build resource to use.
   - `spec.serviceAccount` - Refers to the SA to use when building the image. (_defaults to the `default` SA_)
   - `spec.timeout` - Defines a custom timeout. The value needs to be parsable by [ParseDuration](https://golang.org/pkg/time/#ParseDuration), for example, `5m`. The value overwrites the value that is defined in the `Build`.
   - `spec.paramValues` - Refers to a name-value(s) list to specify values for `parameters` defined in the `BuildStrategy`. This value overwrites values defined with the same name in the Build.
   - `spec.output.image` - Refers to a custom location where the generated image would be pushed. The value will overwrite the `output.image` value defined in `Build`. ( Note: other properties of the output, for example, the credentials, cannot be specified in the buildRun spec. )
-  - `spec.output.credentials.name` - Reference an existing secret to get access to the container registry. This secret will be added to the service account along with the ones requested by the `Build`.
+  - `spec.output.pushSecret` - Reference an existing secret to get access to the container registry. This secret will be added to the service account along with the ones requested by the `Build`.
   - `spec.env` - Specifies additional environment variables that should be passed to the build container. Overrides any environment variables that are specified in the `Build` resource. The available variables depend on the tool used by the chosen build strategy.
 
-_Note:_ The `BuildRef` and `BuildSpec` are mutually exclusive. Furthermore, the overrides for `timeout`, `paramValues`, `output`, and `env` can only be combined with `buildRef`, but **not** with `buildSpec`.
```

Comment:
@qu1queee @SaschaSchwarze0 AFAIU, with v1beta1, the `BuildSpec` & `BuildRef` are now replaced with `spec.build.spec` & `spec.build.name` for buildRun.
Does the above statement still apply for both (build.spec & build.name being mutually exclusive)?

Reply:
As per the slack discussion, I got the answer to the above i.e., YES. One can either specify `spec.build.spec` or `spec.build.name`. Thanks!
 
A follow-up que:  What are we indicating with the following statement? Is it that the specified parameters can still be overridden through some other CR when we use `spec.build.name`, and the parameters specified in the fields of `spec.build.spec` will remain intact as specified in the buildRun?
~~~
 Furthermore, the overrides for `timeout`, `paramValues`, `output`, and `env` can only be combined with `buildRef`, but **not** with `buildSpec`.
~~~

Reply:
> A follow-up que: What are we indicating with the following statement? Is it that the specified parameters can still be overridden through some other CR when we use spec.build.name, and the parameters specified in the fields of spec.build.spec will remain intact as specified in the buildRun?

Yes